day2/lib/day2.rb

```diff
@@ -0,0 +1,41 @@
+module Day2
+  class PasswordList
+    attr_accessor :password_list, :entry_counts
+
+    def initialize(password_list)
+      @password_list = password_list
+    end
+
+    def self.from_file(filepath)
+      new(File.foreach(filepath).map { |line| line.chomp })
+    end
+
+    def count_valid_entries(entry_klass)
+      @password_list.map { |raw_entry| entry_klass.new(raw_entry).valid? }.count(true)
+    end
+  end
+
+  class Part1
+    class Entry
+      attr_reader :min_chars, :max_chars, :char, :password
+
+      def initialize(raw_entry)
+        char_limits, raw_char, @password = raw_entry.split(" ")
+        @min_chars, @max_chars = char_limits.split("-").map { |raw_char| raw_char.to_i }
```

Comment:
```suggestion
        @min_chars, @max_chars = char_limits.split("-").map(&:to_i)
```

Reply:
Huge fan. Applied it here and two other places via 78ff478528788bfba80dfc8d8d87a6413e0faca3.